app/src/main/java/com/github/geohunt/app/model/database/firebase/FirebaseUser.kt

```diff
@@ -20,11 +23,32 @@ class FirebaseUser(
     override val hunts: List<LazyRef<Challenge>>,
     override val numberOfFollowers: Int,
     override val follows: List<LazyRef<User>>,
-    override var score: Long
+    override var score: Long,
+    override val isPOIUser: Boolean = false,
 ) : User {
 
 }
 
+/**
+ * Internal implementation of the LazyRef for the firebase user POI-User specifically
+ */
+internal class FirebasePOIUserRef(override val id: String) :
+    BaseLazyRef<User>() {
+    override fun fetchValue(): Task<User> {
+        return Tasks.forResult(FirebaseUser(
+            uid = id,
+            displayName = null,
+            profilePicture = InvalidLazyRef(RuntimeException("Cannot fetch profile picture for POI User")),

```

Comment:
Throwing an exception is probably not the best way. Instead, maybe returning a constant Bitmap?

Reply:
After thinking about it, it is not a good idea. The id of this bitmap won't correspond to anything and introducing a "second" mechanism to fetch bitmap is just complexification of the codebase. Beside I do not expect this special POI-User to be rendered. As such the UI should not display a user when it is POI